eras/mary/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -292,28 +306,20 @@ encodeMultiAssetMaps ::
   Encoding
 encodeMultiAssetMaps (MultiAsset m) = toCBOR m
 
-decodeMultiAssetMaps ::
-  Crypto c =>
-  Decoder s Integer ->
-  Decoder s (MultiAsset c)
+-- | `MultiAsset` can be used in two different circumstances:
+-- during minting and in `MaryValue` while sending.
+-- In the latter case, MultiAsset amounts cannot be negative,
+-- while in minting negative indicates burning, and should not be zero.
+decodeMultiAssetMaps :: Crypto c => (forall t. Decoder t Integer) -> Decoder s (MultiAsset c)
 decodeMultiAssetMaps decodeAmount =
-  MultiAsset . prune <$> decodeMap fromCBOR (decodeMap fromCBOR decodeAmount)
+  MultiAsset <$> decodeMap fromCBOR (decodeMap fromCBOR decodeAmount)
```

Comment:
@lehins if `prune` is retained here, the tests throw `No Decoder error` `No Dropper error`. This is most likely due to cases where there are `0` or `[]`.

Reply:
Yeah, that's right, we don't want to prune zero values, we want to fail on those. However that brings up another point. We were before pruning empty maps too, so we need to preserve this behavior, but also I think we need to fail whenever empty map is supplied starting with Conway.

@JaredCorduan What's your opinion here. Do we continue to prune empty maps or fail when one is supplied? I lean towards the latter

Reply:
I don't have a strong opinion about this. If we do want to disallow empty maps, let's update the CDDL spec to use `+` instead of `*`.  so I think we'd want `multiasset<a> = { + policy_id => { + asset_name => a } }` (ie no empty map of policies, and no empty map of assets within a given policy.

Reply:
If we are not pruning,  we need to guard against them
```suggestion
    ma <- decodeMap fromCBOR (decodeMap fromCBOR decodeAmount)
    ifDecoderVersionAtLeast
    (natVersion @9)
    (MultiAsset ma <$ mapM_ ma (\m -> when (Map.empty m) $ fail "Empty Assets are not allowed")
    (pure $ MultiAsset $ prune ma)
```

Reply:
Note, that unlike the spec we do allow multiasset to be empty, because otherwise we'd have to make `mint` field and mulit asset part of the MaryValue be `Maybe MultiAsset`, instead we treating empty as `Nothing`

Reply:
yep, great point @lehins. fortunately, I do not have any evidence that this will cause any confusion for folks following the chain making guesses about the semantics.